implementations/rust/ockam/ockam_transport_tcp/src/portal/portal_worker.rs

```diff
@@ -620,6 +632,10 @@ impl Worker for TcpPortalWorker {
 
                 if remote_packet {
                     let msg = PortalMessage::decode(&payload)?;
+                    if !msg.is_disconnect() {
+                        let span = self.start_span(ctx, &their_identifier)?;
```

Comment:
A general question about spans. Once a span is created, it's handled by tracing? You don't need to care about it being dropped and stuff like that right?

Reply:
When a span is created as above it is started it with a specific tracer (the global one) and the tracer takes care of it.
This is a bit different if you use the `tracing` crate directly because then you have to make sure you enter the span, and then when it is dropped, it will get an end timestamp. 

Your question makes me realize that in the code above we don't end the span. Since we don't track request/responses (we get an independent trace for the response message in a request/response scenario), we don't ever end the span. The better thing would be to end the current span when a transport message leaves the node. At least that would give us an idea of how long it took to process a message on a given node. I will have a look at that later when I come back to improving our data.